networkx/algorithms/richclub.py

```diff
@@ -75,7 +75,8 @@ def rich_club_coefficient(G, normalized=True, Q=100, seed=None):
             "rich_club_coefficient is not implemented for graphs with self loops."
         )
     rc = _compute_rc(G)
-    if normalized:
+    # need at least 4 nodes to perform a double edge swap
+    if normalized and G.number_of_nodes() >= 4:
```

Comment:
I don't think you need to check `G.number_of_nodes() >= 4` here. Currently, the `nx.double_edge_swap` raises a  `NetworkXError: Graph has fewer than four nodes.` for this case.

```.py
G = nx.Graph()
G.add_nodes_from([0,1,2])
G.add_edges_from([(1,2),(2,0)])
print(nx.rich_club_coefficient(G))
 
# output : 
# currently : NetworkXError: Graph has fewer than four nodes.
# now : {0: 0.6666666666666666}
```

Reply:
Right, I tried to make the call exception-free by defaulting to non-normalized if there are less than 4 nodes. If this is not the intended way, we can change it back. (Though issue #6920 was a crash report, I'm guessing the issue reporter preferred better crash handling, unless it's the function-caller's responsibility to ensure >=4 nodes when calling with normalized=True). What do you suggest?

Reply:
Here this NetworkXError is not a crash or strange result, because it clearly states why the error is raised in the error msg. The normalization can be adjusted by the user in the function call by setting `normalized=False`. If you want to change the default behavior of the  `normalized` parameter(instead of raising an exception, like right now) then you would need to deprecate this function's behavior.